h11/_writers.py

```diff
@@ -95,23 +108,32 @@ def send_data(self, data, write):
         write(data)
         write(b"\r\n")
 
-    def send_eom(self, headers, write):
+    def send_eom(self, headers: Headers, write: Writer) -> None:
         write(b"0\r\n")
         write_headers(headers, write)
 
 
 class Http10Writer(BodyWriter):
-    def send_data(self, data, write):
+    def send_data(self, data: bytes, write: Writer) -> None:
         write(data)
 
-    def send_eom(self, headers, write):
+    def send_eom(self, headers: Headers, write: Writer) -> None:
         if headers:
             raise LocalProtocolError("can't send trailers to HTTP/1.0 client")
         # no need to close the socket ourselves, that will be taken care of by
         # Connection: close machinery
 
 
-WRITERS = {
+WritersType = Dict[
+    Union[Tuple[Sentinel, Sentinel], Sentinel],
+    Union[
+        Dict[str, Type[BodyWriter]],
+        Callable[[Union[InformationalResponse, Response], Writer], None],
+        Callable[[Request, Writer], None],
+    ],
+]
```

Comment:
I think this typing here helps illuminate that the `WRITERS` usage here is a bit odd, with it's special case for `SEND_BODY`. 😀
Not actionable tho - just a passing observation.